### PR TITLE
Feature/simd neon

### PR DESCRIPTION
**Pull request by gMerm**
First commit of the program using NEON intrinsics.

### CODE REVIEW

File: average_neon.c [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+        //load the corresponding 4 values from the loop, perform addition & store result in vec
+        float32x4_t input = vld1q_f32(&A[i]);
+        vec=vaddq_f32(vec,input);
+    }
+    
+    
+    //declare at as 32x2 because it will store only 2 floats after the addition takes place
+    //extract the 2 upper floats and the 2 lower floats and add them
+    float32x2_t upper_half=vadd_f32(vget_high_f32(vec), vget_low_f32(vec));
```
Comment (markos) [inline average_neon.c]: very nicely done, but I will suggest you look at the intrinsic vaddvq_f32() and see the difference.

Comment (gMerm) [reply average_neon.c]: Thank you, I will take a look. I'm already making some corrections.

**Comment by markos** [comment]
Very nice, fix the tabs that look a bit off and it's good to be merged!

**Comment by gMerm** [comment]
> Very nice, fix the tabs that look a bit off and it's good to be merged!

Thank you, I'll fix the tabs you mentioned and I'll add a Makefile.

**Comment by markos** [comment]
Some compiler warnings:
```
$ make
gcc -O3 -march=native -Wall -o average_neon average_neon.c 
average_neon.c: In function ‘main’:
average_neon.c:84:35: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘__suseconds_t’ {aka ‘long int’} [-Wformat=]
   84 |   printf("scalar: %ld sec, usec: %d\n", diff1.tv_sec, diff1.tv_usec);
      |                                  ~^                   ~~~~~~~~~~~~~
      |                                   |                        |
      |                                   int                      __suseconds_t {aka long int}
      |                                  %ld
average_neon.c:86:36: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘__suseconds_t’ {aka ‘long int’} [-Wformat=]
   86 |   printf("NEON   : %ld sec, usec: %d\n", diff2.tv_sec, diff2.tv_usec);
      |                                   ~^                   ~~~~~~~~~~~~~
      |                                    |                        |
      |                                    int                      __suseconds_t {aka long int}
      |                                   %ld
average_neon.c:17:3: warning: ignoring return value of ‘posix_memalign’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   17 |   posix_memalign((void **)&A, 16, N * sizeof(float));
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
average_neon.c:89:3: warning: ‘A’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   89 |   free(A);
      |   ^~~~~~~
```

in general posix_memalign() -and other alloc functions should not be ignored.

**Review by markos — APPROVED** [review]
LGTM

**Pull request merged**